File: src/main/java/toast/pine/Entities/EntityManager.java
package toast.pine.Entities;

import org.bukkit.entity.LivingEntity;
import toast.pine.Events.MonsterDeathEvent;
import toast.pine.Events.MonsterSpawnEvent;
import toast.pine.Events.MonsterTargetPlayerEvent;

import java.util.HashMap;
import java.util.Map;

public class EntityManager {

    private final Map<LivingEntity, EntityHandler> activeLivingEntities = new HashMap<>();

    public void registerHandledEntity(LivingEntity Entity, EntityHandler handler) {
        this.activeLivingEntities.put(Entity, handler);
    }

    public void unregisterHandledEntity(LivingEntity entity) {
        this.activeLivingEntities.remove(entity);
    }


    public void handleMonsterSpawn(MonsterSpawnEvent event) {
        EntityHandler handler = this.activeLivingEntities.get(event.getEntity());
        if (handler != null) {
            handler.onMonsterSpawn(event);
        }
    }

    public void handleMonsterDeath(MonsterDeathEvent event){
        EntityHandler handler = this.activeLivingEntities.get(event.getEntity());
        if (handler != null) {
            handler.onMonsterDeath(event);
        }
    }

    public void handlerMonsterTarget(MonsterTargetPlayerEvent event) {
        EntityHandler handler = this.activeLivingEntities.get(event.getMonster().getLivingEntity());
        if (handler != null) {
            handler.onMonsterTarget(event);
        }
    }





}

File: src/main/java/pine/toast/Entities/EntityManager.kt
package toast.pine.Entities

import org.bukkit.entity.LivingEntity
import toast.pine.Events.MonsterDeathEvent
import toast.pine.Events.MonsterSpawnEvent
import toast.pine.Events.MonsterTargetPlayerEvent

class EntityManager {
    private val activeLivingEntities: MutableMap<LivingEntity, EntityHandler> = HashMap()


    fun registerHandledEntity(entity: LivingEntity, handler: EntityHandler) {
        activeLivingEntities[entity] = handler
    }

    fun unregisterHandledEntity(entity: LivingEntity) {
        activeLivingEntities.remove(entity)
    }

    fun handleMonsterSpawn(event: MonsterSpawnEvent) {
        val handler = activeLivingEntities[event.entity]
        handler?.onMonsterSpawn(event)
    }

    fun handleMonsterDeath(event: MonsterDeathEvent) {
        val handler = activeLivingEntities[event.getEntity()]
        handler?.onMonsterDeath(event)
    }

    fun handlerMonsterTarget(event: MonsterTargetPlayerEvent) {
        val handler = activeLivingEntities[event.getMonster().getLivingEntity()]
        handler?.onMonsterTarget(event)
    }
}
